base/test/launcher/teamcity_service_messages.cc

```diff
@@ -0,0 +1,119 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/base/test/launcher/teamcity_service_messages.h"
+
+#include <string>
+
+#include "base/ranges/algorithm.h"
+#include "base/strings/string_number_conversions.h"
+
+namespace base {
+namespace {
+
+// https://www.jetbrains.com/help/teamcity/service-messages.html#Escaped+Values
+std::string_view TeamcityValueEscape(std::string_view s, std::string& result) {
+  static constexpr char kSymbolsToEscape[] = "\n\r'|[]";
+  const char* s_char = ranges::find_first_of(s, kSymbolsToEscape);
+  if (s_char == s.end()) {
+    return s;
```

Comment:
Could me integrate that case in the next `for + switch` block to avoid syncing `kSymbolsToEscape` with the switch cases?
Maybe it's fine to append chars one by one to ostream. It isn't look like a perf bottleneck.